cyclonedx-bom/src/external_models/uri.rs

```diff
@@ -60,6 +60,12 @@ impl Validate for Purl {
     }
 }
 
+impl<'a> From<PackageUrl<'a>> for Purl {
```

Comment:
The PackageUrl crate currently has version 0.3.0, not 1.0 or something like that. It is likely that they will bump the version frequently. If we provide a `From<PackageUrl>`  implementation, that will expose the `PackageUrl` type in the public API, and require an incompatible SemVer change to this crate every time the SemVer of `PackageUrl` is bumped.

Could you provide a `FromStr` implementation instead?